src/encoding.ts

```diff
@@ -0,0 +1,337 @@
+type ByteBuffer = Uint8Array | Buffer | number[];
+const BASE64_KEYSTR = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/=';
+const HIGH_SURROGATE_MIN = 0xD800;
```

Comment:
My original suggestion was to put these constants in `string.ts` then import them here (and also use them there).